salt/modules/mount.py

```diff
@@ -745,19 +751,19 @@ def set_vfstab(
 
     except (IOError, OSError) as exc:
         msg = 'Couldn\'t read from {0}: {1}'
-        raise CommandExecutionError(msg.format(config, str(exc)))
+        raise CommandExecutionError(msg.format(config, six.text_type(exc)))
 
     # add line if not present or changed
     if ret is None:
-        lines.append(str(entry))
+        lines.append(six.text_type(entry))
         ret = 'new'
 
     if ret != 'present':  # ret in ['new', 'change']:
         if not salt.utils.args.test_mode(test=test, **kwargs):
             try:
                 with salt.utils.files.fopen(config, 'w+') as ofile:
                     # The line was changed, commit it!
-                    ofile.writelines(lines)
+                    ofile.writelines(salt.utils.data.decode(lines))
```

Comment:
Same as above, this needs to be an `encode`, and we need to open the file using `'wb'`.